.adr/0014-update-event-emission.md

```diff
@@ -0,0 +1,38 @@
+# 0014 Channel Update Event Emission
+
+## Status
+
+Review
+
+## Context
+
+We want to enhance the our go-nitro RPC client and server with the ability to receive notifications whenever a channel is updated. This allows us to design a UI that can easily update when a channel changes.
+
+Previously in [ADR 0012](./0012-event-emission.md) we decided to `close` channels to notify consumers that an objective have been completed. Using `close` allowed us to broadcast a signal to multiple consumers at once. While this works well for the objective completed event, it does not work for channel updates. This is because:
+
+1. `close` doesn't allow us to pass any information (like channel data)
+2. `close` can only be used once as a signal on a channel.
+
+To support channel updates we need another approach that let's us dispatch channel updates to multiple consumers.
+
+## Decision
+
+We will implement the "Slice of chans" pattern similiar to the one outlined in [ADR 0012](./0012-event-emission.md).
+
+The nitro client will store a slice of event listener `chans` for each channel id. A consumer can "subscribe" by calling a method which appends a `chan` to that slice and returns it. Whenever the client receives a channel update from the engine it iterates on over the slice and sends an update to any event listener `chans`.
+
+To prevent duplicate notifications being emitted, the client keeps track of the previous emitted channel update. A channel update notification is only emitted if the channel has changed from the previous emitted notification.
+
+## Alternatives considered
+
+### Using a Third Party Library
+
+We could use a third-party library, like [EventBus](https://github.com/asaskevich/EventBus), to handle the work of pubsub. However by writing our own eventhandler logic we can adjust it to our needs. One example is comparing the new event against the previously emitted event to prevent duplicate events being fired.
+
+### Sync.Cond
+
+While `sync.Cond` allows for broadcasting updates to multiple consumers, it requires the consumers and broadcaster to be goroutines. This means we'd have to spin up goroutines for each event listener which does not seem like a natural fit,.
```

Comment:
There's something that doesn't quite sit right with me here. 

In my head, listeners could quite naturally be go routines? When a listener subscribes, they get a chan which they can essentially use to trigger a callback by spawning a goroutine which first reads from the chan (and then does some action they want to be triggered by the event). If we were to use a `c := sync.Cond` , the listener can call `c.Wait()` to block until the event is fired and there's a similar pattern. 

Perhaps you have in mind that there are other patterns (maybe a `select` statement such as the one used in `checkNotifications`) which won't work so well with a `sync.Cond`? It seems to me that one can most likely work around that (you would still need goroutines, though).

Reply:
Thanks. Good point about not being able to communicate any data with `.Broadcast()`. 

I've taken the liberty of updating our discussion on `sync.Cond` across both ADRs [2f2d5c3](https://github.com/statechannels/go-nitro/pull/1250/commits/2f2d5c3e120ed84a38cb757a71b689580d400362).